kroxylicious-operator/src/test/java/io/kroxylicious/kubernetes/operator/LocallyRunningOperatorRbacHandler.java

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.operator;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.UUID;
+
+import org.junit.jupiter.api.AutoClose;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.KubernetesResourceList;
+import io.fabric8.kubernetes.api.model.rbac.ClusterRole;
+import io.fabric8.kubernetes.api.model.rbac.ClusterRoleBinding;
+import io.fabric8.kubernetes.api.model.rbac.ClusterRoleBindingBuilder;
+import io.fabric8.kubernetes.api.model.rbac.ClusterRoleBuilder;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClientBuilder;
+import io.fabric8.kubernetes.client.dsl.EditReplacePatchable;
+import io.fabric8.kubernetes.client.dsl.NonNamespaceOperation;
+import io.fabric8.kubernetes.client.dsl.Resource;
+import io.javaoperatorsdk.operator.junit.AbstractOperatorExtension;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+/**
+ * Designed to cooperate with {@link io.javaoperatorsdk.operator.junit.LocallyRunOperatorExtension}
+ * so that the operator runs locally with role based access restrictions as close to that it will
+ * have when deployed to Kubernetes for real.
+ * <br/>
+ * Register this extension before LocallyRunOperatorExtension, passing LocallyRunOperatorExtension
+ * the kubernetes client returned by {@link #operatorClient}.  This client will use a principal
+ * which has been configured in Kubernetes to have the same RBAC rights as the operator will use
+ * in production.
+ * <br/>
+ * For the test's interactions with Kubernetes, where more privileges are required, use the operations
+ * exposed by {@link #testActor(AbstractOperatorExtension)}.
+ *
+ */
+public class LocallyRunningOperatorRbacHandler implements BeforeEachCallback, AfterEachCallback {
```

Comment:
I originally tried a Reconciliation IT abstract base class.   That approach doesn't work because of the order Junit calls before/afters.  Basically, LocallyRunningOperatorRbacHandler need s be an extension that is registered before LocallyRunOperatorExtension.